### ISSUE TITLE
vomeip examples not making a function call 

### ISSUE THREAD

**Issue by hanmhs9** [COVESA/vsomeip#417]
Hi, I am basically having a same issue with [#56 ](github.com/COVESA/vsomeip/issues/56)

It works with hello world example, (send world and receive hello world), but for the example posted in the web, the client detects the service being available, but it doesn't send a service request. 

The issue #56 is closed without any solution, so I would appreciate if anyone can help on this. 

Here's the json files and logs on each side

**service.json**

{
    "unicast" : "172.17.0.4",
    "logging" :
    {
        "level" : "debug",
        "console" : "true",
        "file" : { "enable" : "false", "path" : "/tmp/vsomeip.log" },
        "dlt" : "false"
    },
    "applications" :
    [
        {
            "name" : "World",
            "id" : "0x1212"
        }
    ],
    "services" :
    [
        {
            "service" : "0x1234",
            "instance" : "0x5678",
            "unreliable" : "30509"
        }
    ],
    "routing" : "World",
    "service-discovery" :
    {
        "enable" : "true",
        "multicast" : "224.224.224.245",
        "port" : "30490",
        "protocol" : "udp",
        "initial_delay_min" : "10",
        "initial_delay_max" : "100",
        "repetitions_base_delay" : "200",
        "repetitions_max" : "3",
        "ttl" : "3",
        "cyclic_offer_delay" : "2000",
        "request_response_delay" : "1500"
    }
}

**Client.json** 
{
    "unicast" : "172.17.0.5",
    "logging" :
    {
        "level" : "debug",
        "console" : "true",
        "file" : { "enable" : "false", "path" : "/var/log/vsomeip.log" },
        "dlt" : "false"
    },
    "applications" :
    [
        {
            "name" : "Hello",
            "id" : "0x1346"
        }
    ],
    "routing" : "Hello",
    "service-discovery" :
    {
        "enable" : "true",
        "multicast" : "224.224.224.245",
        "port" : "30490",
        "protocol" : "udp",
        "initial_delay_min" : "10",
        "initial_delay_max" : "100",
        "repetitions_base_delay" : "200",
        "repetitions_max" : "3",
        "ttl" : "3",
        "cyclic_offer_delay" : "2000",
        "request_response_delay" : "1500"
    }
}

**service logs:**
VSOMEIP_CONFIGURATION=../tutorial/service.json VSOMEIP_APPLICATION_NAME=World ../tutorial/service-tutorial
2023-03-13 01:58:02.922110 [info] Parsed vsomeip configuration in 0ms
2023-03-13 01:58:02.922826 [info] Using configuration file: "../tutorial/service.json".
2023-03-13 01:58:02.922948 [info] Initializing vsomeip application "World".
2023-03-13 01:58:02.923537 [info] Instantiating routing manager [Host].
2023-03-13 01:58:02.924053 [info] create_local_server Routing endpoint at /tmp/vsomeip-0
2023-03-13 01:58:02.924721 [info] Service Discovery enabled. Trying to load module.
2023-03-13 01:58:02.927307 [info] Service Discovery module loaded.
2023-03-13 01:58:02.928054 [info] Application(World, 1212) is initialized (11, 100).
2023-03-13 01:58:02.928725 [info] OFFER(1212): [1234.5678:0.0] (true)
2023-03-13 01:58:02.929723 [info] Listening at /tmp/vsomeip-1212
2023-03-13 01:58:02.930279 [info] Starting vsomeip application "World" (1212) using 2 threads I/O nice 255
2023-03-13 01:58:02.931955 [info] main dispatch thread id from application: 1212 (World) is: 7f1b9ae5e700 TID: 550
2023-03-13 01:58:02.932139 [info] shutdown thread id from application: 1212 (World) is: 7f1b9a65d700 TID: 551
2023-03-13 01:58:02.933380 [info] Watchdog is disabled!
2023-03-13 01:58:02.936050 [info] io thread id from application: 1212 (World) is: 7f1b9aecc000 TID: 549
2023-03-13 01:58:02.935959 [info] io thread id from application: 1212 (World) is: 7f1b9965b700 TID: 553
2023-03-13 01:58:02.937318 [info] vSomeIP 3.1.20.3 | (default)
2023-03-13 01:58:02.938838 [info] Network interface "eth0" state changed: up
2023-03-13 01:58:02.939916 [info] Route "default route (0.0.0.0/0) if: eth0 gw: 172.17.0.1" state changed: up
2023-03-13 01:58:02.940832 [info] udp_server_endpoint_impl: SO_RCVBUFFORCE successful.
2023-03-13 01:58:02.941033 [info] udp_server_endpoint_impl: SO_RCVBUF is: 1703936 (1703936) local port:30490
2023-03-13 01:58:02.941244 [debug] Joining to multicast group 224.224.224.245 from 172.17.0.4
2023-03-13 01:58:02.941798 [info] udp_server_endpoint_impl<multicast>: SO_RCVBUFFORCE: successful.
2023-03-13 01:58:02.942239 [info] udp_server_endpoint_impl<multicast>: SO_RCVBUF is: 1703936 (1703936) local port:30490
2023-03-13 01:58:02.943365 [info] udp_server_endpoint_impl: SO_RCVBUFFORCE successful.
2023-03-13 01:58:02.943852 [info] udp_server_endpoint_impl: SO_RCVBUF is: 1703936 (1703936) local port:30509
2023-03-13 01:58:02.944437 [info] SOME/IP routing ready.
2023-03-13 01:58:02.945039 [warning] Route "224.224.224.245/32 if: eth0 gw: n/a" state changed: up
^C2023-03-13 01:58:11.581448 [info] Stopping vsomeip application "World" (1212).

**client log** 
../tutorial/client.json VSOMEIP_APPLICATION_NAME=Hello ../tutorial/client-tutorial
2023-03-13 01:57:56.814084 [info] Parsed vsomeip configuration in 0ms
2023-03-13 01:57:56.814947 [info] Using configuration file: "../tutorial/client.json".
2023-03-13 01:57:56.815070 [info] Initializing vsomeip application "Hello".
2023-03-13 01:57:56.816230 [info] Instantiating routing manager [Host].
2023-03-13 01:57:56.816998 [info] create_local_server Routing endpoint at /tmp/vsomeip-0
2023-03-13 01:57:56.818057 [info] Service Discovery enabled. Trying to load module.
2023-03-13 01:57:56.820596 [info] Service Discovery module loaded.
2023-03-13 01:57:56.821813 [info] Application(Hello, 1346) is initialized (11, 100).
2023-03-13 01:57:56.822292 [info] REQUEST(1346): [1234.5678:255.4294967295]
2023-03-13 01:57:56.823676 [info] Listening at /tmp/vsomeip-1346
2023-03-13 01:57:56.825089 [info] Starting vsomeip application "Hello" (1346) using 2 threads I/O nice 255
2023-03-13 01:57:56.826229 [info] main dispatch thread id from application: 1346 (Hello) is: 7f3b38534700 TID: 643
2023-03-13 01:57:56.827656 [info] shutdown thread id from application: 1346 (Hello) is: 7f3b37d33700 TID: 644
2023-03-13 01:57:56.829057 [info] Watchdog is disabled!
CLIENT: Service [1234.5678] is NOT available.
2023-03-13 01:57:56.830847 [info] io thread id from application: 1346 (Hello) is: 7f3b38da2ec0 TID: 641
**2023-03-13 01:57:56.831568 [error] Routing info for remote service could not be found! (1346): [1234.5678.0421] 0001**
2023-03-13 01:57:56.831523 [info] io thread id from application: 1346 (Hello) is: 7f3b36d31700 TID: 646
2023-03-13 01:57:56.832019 [info] vSomeIP 3.1.20.3 | (default)
2023-03-13 01:57:56.832943 [info] Network interface "eth0" state changed: up
2023-03-13 01:57:56.834421 [info] Route "default route (0.0.0.0/0) if: eth0 gw: 172.17.0.1" state changed: up
2023-03-13 01:57:56.835713 [info] udp_server_endpoint_impl: SO_RCVBUFFORCE successful.
2023-03-13 01:57:56.836476 [info] udp_server_endpoint_impl: SO_RCVBUF is: 1703936 (1703936) local port:30490
2023-03-13 01:57:56.836804 [debug] Joining to multicast group 224.224.224.245 from 172.17.0.5
2023-03-13 01:57:56.837336 [info] udp_server_endpoint_impl<multicast>: SO_RCVBUFFORCE: successful.
2023-03-13 01:57:56.837452 [info] udp_server_endpoint_impl<multicast>: SO_RCVBUF is: 1703936 (1703936) local port:30490
2023-03-13 01:57:56.837997 [info] SOME/IP routing ready.
2023-03-13 01:57:56.838625 [warning] Route "224.224.224.245/32 if: eth0 gw: n/a" state changed: up
2023-03-13 01:58:02.732287 [info] endpoint_manager_impl::create_remote_client: 172.17.0.4:30509 reliable: 0 using local port: 0
2023-03-13 01:58:02.733105 [info] udp_client_endpoint_impl::connect: SO_RCVBUFFORCE successful!
2023-03-13 01:58:02.733221 [info] udp_client_endpoint_impl::connect: SO_RCVBUF is: 1703936 (1703936) local port:0 remote:172.17.0.4:30509
**CLIENT: Service [1234.5678] is available.**
2023-03-13 01:58:06.839162 [info] vSomeIP 3.1.20.3 | (default)
CLIENT: Service [1234.5678] is NOT available.
^C2023-03-13 01:58:11.978091 [info] Stopping vsomeip application "Hello" (1346).
2023-03-13 01:58:11.985931 [info] Exiting vsomeip application...

Basically it detects the service but it doesn't send request. 

Please help!

**Comment by hanmhs9** [comment]
The problem solved by modifying the line to **if(_is_available) condition.notify_one();**  instructed by rvashi2020 in[ issue#44 ](https://github.com/COVESA/vsomeip/issues/44)solved the problem
